internal/adminapi/endpoints.go

```diff
@@ -58,10 +64,10 @@ func GetURLsForService(
 	return addresses, nil
 }
 
-// AddressesFromEndpointSlice returns a list of Admin API addresses when given
-// an Endpointslice.
-func AddressesFromEndpointSlice(endpoints discoveryv1.EndpointSlice, portNames sets.Set[string]) sets.Set[string] {
-	addresses := sets.New[string]()
+// AdminAPIsFromEndpointSlice returns a list of Admin APIs when given
+// an EndpointSlice.
+func AdminAPIsFromEndpointSlice(endpoints discoveryv1.EndpointSlice, portNames sets.Set[string]) sets.Set[DiscoveredAdminAPI] {
```

Comment:
Just a small nit: the package is called `adminapi` so using this as `adminapi.AdminAPIsFromEndpointSlice()` will stutter.